content/articles/json-server-and-axios/index.md

```diff
@@ -0,0 +1,398 @@
+
+#Inside your working directory, start the JSON Server
+$ json-server data.json
+```
+
+If we type <http://localhost:3000/items> in our browser, we should see our shopping list in the database.
+
+### Getting HTTP data within our application
+First, we need to install `axios` in our project to access the data in the `data.json` file:
+
+```bash
+npm install axios
+```
+
+After the installation, we can now import `axios` in our application. 
+
+In our `App.vue`,let's remove everything inside template, script and style tags. Add the following code:
+
+```vuejs
+<template>
+  <div id="app">
+    <h1>Shopping List</h1>
+  </div>
+  <ul>
+    <li v-for="item of items" :key="item.id">
+      {{ item.name }}
+    </li>
+  </ul>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  name: "App",
+  data() {
+    return {
+      items: [],
+    };
+  },
+  async created() {
+    try {
+      const res = await axios.get(`http://localhost:3000/items`);
+      this.items = res.data;
+    } catch (error) {
+      console.log(error);
+    }
+  },
+};
+</script>
+
+<style>
+#app {
+  text-align: center;
+  color: #2c3e50;
+  margin-top: 60px;
+}
+
+li {
+  list-style: none;
+}
+</style>
+```
+
+From the above code, we have imported Axios and utilized the [async and await](https://www.javascripttutorial.net/es-next/javascript-async-await/) keywords, which build on promises and allow you to construct asynchronous code.
+
+### Adding data using POST
+You can make `POST`, `PUT`, `PATCH` or `DELETE` requests to the database using Axios.
+
+`POST` is a HTTP method used to create or add data in the database. In our project, let's add `POST` method to add an item in our shopping list:
+
+```vuejs
+<template>
+  <div id="app">
+    <h1>Shopping List</h1>
+    <input v-model="itemName" type="text" /><br />
+    <button @click="addItem()">Add Item</button>
+  </div>
+  <ul>
+    <li v-for="item of items" :key="item.id">
+      {{ item.name }}
+    </li>
+  </ul>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  name: "App",
+  data() {
+    return {
+      items: [],
+      itemName: "",
+    };
+  },
+  async created() {
+    try {
+      const res = await axios.get(`http://localhost:3000/items`);
+      this.items = res.data;
+    } catch (error) {
+      console.log(error);
+    }
+  },
+  methods: {
+    async addItem() {
+      const res = await axios.post(`http://localhost:3000/items`, {
+        name: this.itemName,
+      });
+      this.items = [...this.items, res.data];
+      this.itemName = "";
+    },
+  },
+};
+</script>
+
+<style>
+#app {
+  text-align: center;
+  color: #2c3e50;
+}
+
+li {
+  list-style: none;
+}
+
+button {
+  margin-top: 5px;
+  background-color: #3498db;
+  border: none;
+  color: #ffffff;
+  padding: 10px 20px;
+  font-size: 14px;
+  cursor: pointer;
+  border-radius: 4px;
+}
+
+input {
+  margin-top: 5px;
+  padding: 10px 20px;
+  font-size: 14px;
+  border-radius: 4px;
+}
+</style>
+
```

Comment:
If possible, break down this code snippet. You can move the stylings to a new snippet.

If needed, you can just add GitHub repo link, since there's nothing much to explain in stylings.